server/src/main/java/org/elasticsearch/action/admin/indices/resolve/ResolveIndexAction.java

```diff
@@ -528,22 +528,33 @@ protected void doExecute(Task task, Request request, final ActionListener<Respon
         static void resolveIndices(
             String[] names,
             IndicesOptions indicesOptions,
-            Metadata metadata,
-            IndexAbstractionResolver resolver,
+            ClusterState clusterState,
+            IndexNameExpressionResolver resolver,
             List<ResolvedIndex> indices,
             List<ResolvedAlias> aliases,
-            List<ResolvedDataStream> dataStreams,
-            boolean includeDataStreams
+            List<ResolvedDataStream> dataStreams
         ) {
-            List<String> resolvedIndexAbstractions = resolver.resolveIndexAbstractions(names, indicesOptions, metadata, includeDataStreams);
-            SortedMap<String, IndexAbstraction> lookup = metadata.getIndicesLookup();
+            // redundant check to ensure that we don't resolve the list of empty names to "all" in this context
+            if (names.length == 0) {
+                return;
+            }
+            // TODO This is a dirty hack around the IndexNameExpressionResolver optimisation for "*" as described in:
+            // https://github.com/elastic/elasticsearch/issues/92903.
+            // A standalone "*" expression is resolved slightly differently from a "*" embedded in another expression, eg "idx,*".
+            // The difference is only slight, and it usually doesn't cause problems (see
+            // https://github.com/elastic/elasticsearch/issues/92911 for a description of a problem).
+            // But in the case of the Resolve index API, the difference is observable, because resolving standalone "*" cannot show
+            // aliases (only indices and datastreams). The Resolve index API needs to show the aliases that match wildcards.
+            if (names.length == 1 && (Metadata.ALL.equals(names[0]) || Regex.isMatchAllPattern(names[0]))) {
+                names = new String[] { "**" };
+            }
```

Comment:
Thanks for the detailed context. It is very helpful!
If the buggy behaviour of `*` is only observed security is disabled, does that mean using `**` here is actually a change of bebhaviour when security is disabled? Or do we consider this one of the bug fixes? If so, do you mind list it in the PR description?

It should make no difference when security is enabled, right?

Reply:
> If the buggy behaviour of * is only observed security is disabled, does that mean using ** here is actually a change of bebhaviour when security is disabled? 

Not quite because the original code  didn't use the buggy code for `*` (it used the implementation in `IndexAbstractionResolver`, and the buggy code is in `IndexNameExpressionResolver`).
I'm not aware of any changes of behavior besides the ones already listed in the PR description.

Reply:
You are right. Thanks for correcting me.